src/computeSourceMap.ts

```diff
@@ -33,3 +51,60 @@ export default function computeSourceMap(
     names: [],
   };
 }
+
+function computeSourceColumns(code: string, tokens: Array<Token>): Array<number> {
+  const sourceColumns: Array<number> = new Array(tokens.length);
+  let j = 0;
+  let currentMapping = tokens[j].start;
+  let lineStart = 0;
+  for (let i = 0; i < code.length; i++) {
+    if (i === currentMapping) {
+      sourceColumns[j] = currentMapping - lineStart;
+      currentMapping = tokens[++j].start;
+    }
+    if (code.charCodeAt(i) === charCodes.lineFeed) {
+      lineStart = i + 1;
+    }
+  }
+  return sourceColumns;
+}
+
+function computeDetailedSourceMap(
```

Comment:
Thanks! The logic all looks good to me, and it's nice to have it self-contained here. It might be better at some point to switch to pre-computing token column indices if source maps will always need all of them, but there's certainly a tradeoff, and this approach seems good.